convert2rhel/checks.py

```diff
@@ -835,3 +841,78 @@ def check_dbus_is_running():
         "Could not find a running DBus Daemon which is needed to register with subscription manager.\n"
         "Please start dbus using `systemctl start dbus` or (on CentOS Linux 6), `service messagebus start`"
     )
+
+
+def _is_initramfs_file_valid(filepath):
+    """Internal function to verify if a initramfs file is corrupted.
+
+    This method will rely on using lsinitrd to do the validation. If the
+    lsinitrd returns other value that is not 0, then it means that the file is
+    probably corrupted or may cause problems during the next reboot.
+
+    :param filepath: The path to the initramfs file.
+    :type filepath: str
+    :return: A boolean to determinate if the file is corrupted.
+    :rtype: bool
+    """
+    logger.info("Checking if the '%s' file is not corrupted.", filepath)
+    out, return_code = run_subprocess(
+        cmd=["/usr/bin/lsinitrd", filepath],
+        print_output=False,
+    )
+
+    if return_code != 0:
+        logger.info("Couldn't verify initramfs file. It may be corrupted.")
+        logger.debug("Output of lsinitrd: %s", out)
+        return False
+
+    return True
+
+
+def check_kernel_boot_files():
+    """Check if the required kernel files exists and are valid under the boot partition."""
+    # For Oracle/CentOS Linux 8 the `kernel` is just a meta package, instead,
+    # we check for `kernel-core`. This is not true regarding the 7.* releases.
+    kernel_name = "kernel-core" if system_info.version.major >= 8 else "kernel"
+
+    # Either the package is returned or not. The return_code will be 0 in
+    # either case, so we don't care about checking for that here.
+    output, _ = run_subprocess(["rpm", "-q", "--last", kernel_name], print_output=False)
+
+    # We are parsing the latest kernel installed on the system, which at this
+    # point, should be a RHEL kernel. Since we can't get the kernel version
+    # from `uname -r`, as it requires a reboot in order to take place, we are
+    # detecting the latest kernel by using `rpm` and figuring out which was the
+    # latest kernel installed.
+    latest_installed_kernel = output.split("\n")[0].split(" ")[0]
+    latest_installed_kernel = latest_installed_kernel.split("%s-" % kernel_name)[-1]
+    grub2_config_file = grub.get_grub_config_file()
+    initramfs_file = INITRAMFS_FILEPATH % latest_installed_kernel
+
+    vmlinuz_exists = os.path.exists(VMLINUZ_FILEPATH % latest_installed_kernel)
```

Comment:
Might be worth outputting the result of vmlinuz in a debug log as we do for initramfs

Reply:
We do so already through `The initramfs and vmlinuz files are valid.` info message. I'd say that's sufficient. What else than the information in this message would you like to print in the debug message?

Reply:
It is more so if it is not valid, and that message doesn't say specifically which is bad. Only that one of them are. We log for initramfs in its function but nothing for vmlinuz

Reply:
Something like: 

```suggestion
    latest_installed_kernel = output.split("\n")[0].split(" ")[0]
    latest_installed_kernel = latest_installed_kernel.split("%s-" % kernel_name)[-1]
    grub2_config_file = grub.get_grub_config_file()
    initramfs_file = INITRAMFS_FILEPATH % latest_installed_kernel
    vmlinuz_file = VMLINUZ_FILEPATH % latest_installed_kernel
    
    logger.debug("Checking if the '%s' file exists", vmlinuz_file)
    vmlinuz_exists = os.path.exists(vmlinuz_file)
    logger.debug("Result of vmlinuz presence: %s", vmlinuz_exists)
```

not sure, what you think?

Reply:
> It is more so if it is not valid, and that message doesn't say specifically which is bad. Only that one of them are. We log for initramfs in its function but nothing for vmlinuz

True, good point.

I'd just add a full stop at the end of the messages. We still have a mix of messages with and without a full stop but from time to time I try to make them consistent and have the full stop (e.g. https://github.com/oamg/convert2rhel/pull/501/files).

Also, to be consistent with checking the initramfs file corruption, we can print only the message saying the vmlinuz file is being present (not that it is present).

```python
    logger.debug("Checking if the '%s' file exists.", vmlinuz_file)
    vmlinuz_exists = os.path.exists(vmlinuz_file)
    if not vmlinuz_exists:
        logger.info("The vmlinuz file is not present.")
```

BTW, again to be consistent, we can switch this initramfs message to a debug level:
```python
logger.info("Checking if the '%s' file is not corrupted.", filepath)
```